### DISCUSSION TITLE
modem

### DISCUSSION THREAD

**Discussion by abcd4455663** [86Box/86Box#4305] [Ideas]
add dialup emulation

**Comment by ThatsItForTheOtherOne** [comment]
Modem emulation is already in the latest nightlies.

**Comment by abcd4455663** [comment]
Where, I didn't find it

**Reply by ThatsItForTheOtherOne** [reply]
<img width="800" alt="image" src="https://github.com/86Box/86Box/assets/44384688/97985993-c771-4138-b2b5-546018bdacac">

**Reply by abcd4455663** [reply]
How can I connect to the Internet using it?

**Reply by abcd4455663** [reply]
Driver ?

**Reply by ThatsItForTheOtherOne** [reply]
You dial 0.0.0.0 and connect over SLIP. The modem connects to IPs using the standard Hayes commands. You can also make a phonebook file to associate certain IP addresses with phone numbers.

**Reply by abcd4455663** [reply]
Why doesn't Windows 2000 see the modem?

**Reply by binarymaster** [reply]
Maybe it's not plug-n-play

**Reply by ThatsItForTheOtherOne** [reply]
It's a standard COM port modem. When you set up dialup in Windows, it gives you an option to select which modem it is. Select a generic model and it just works.

**Reply by abcd4455663** [reply]
what is the phonebook file structure?

**Reply by abcd4455663** [reply]
how to write phonebook file?

**Reply by lemondrops** [reply]
It's just a list of phone numbers to dial inside the emulated machine and IP address:port combinations to connect to:
```
12345 192.168.0.1:23
23456 127.0.0.1:20300
```
etc.
Obviously, another 86Box (or DOSBox - since the emulation was ported from there, it's cross-compatible!) instance that's configured to listen on the specified IP:port must be running for the connection to succeed.

**Reply by Vinson8970** [reply]
> It's a standard COM port modem. When you set up dialup in Windows, it gives you an option to select which modem it is. Select a generic model and it just works.

IE 4 doesn't see the modem, it's saying it is in COM2 (and it is), but it won't work.
